crates/types/src/logs/metadata.rs

```diff
@@ -349,6 +349,8 @@ impl From<Logs> for LogsSerde {
     fn from(value: Logs) -> Self {
         Self {
             version: value.version,
+            // As logs is a btreemap, this is ordered by log id, which avoids reordering the entire logs config
+            // on every write.
```

Comment:
Why is this important?

Reply:
I don't know that it's consequential to anything else, but it _is_ nice for operators inspecting the config via `restatectl meta get`.

Reply:
I think this influences the order of the serialized representation in metadata store, and that in turn is nice for operators! I've also wanted to make the same change after trying to monitor logs for changes with `restatectl meta get`. I don't know if there's a good way to only do this in `LogsSerde` rather, and keep the lookup cost of a `HashMap` for programmatic access.

Reply:
Ok, so it is has nothing to do with correctness?

Reply:
correct, its a nice to have. the same may be worth doing for partition table and nodes config etc. its super annoying that when doing meta get, everything is out of order. It makes it much harder to diff before and have 

if we think the lookup cost is relevant (seems extremely unlikely?) i can just sort them when converting to logsserde

Reply:
I think that the look-up costs for `BTreeMap` vs. `HashMap` will probably not be noticeable given the number of logs we are looking at. The one thing that does not feel right to me is that we change the internal representation of the data object to influence how `restatectl` presents the data to users. This feels like an unnecessary coupling which might break in future iterations. For example, a less intrusive change could have been to change the `Display` implementation to sort this set.

Reply:
This should remain as hashmap, for the access pattern we have, the hashmap performed much better in my benchmarks.

Reply:
i can just sort when converting into logsserde. i dont know what restatectl can do as it doesnt understand the nature of the data